### ISSUE TITLE
Epic Games login page blank

### ISSUE THREAD

**Issue by Incognitux** [Heroic-Games-Launcher/HeroicGamesLauncher#3682]
### Describe the bug

When trying to login into epic games from the stores tab it displays nothing but a black screen.

<details>
<summary>Terminal output</summary>

```sh
Software Versions:
  Heroic: 2.14.1 Flame Emperor Sabo
  Legendary: 0.20.34 Direct Intervention
  gogdl: 1.0.1
  Nile: 1.0.2 Jonathan Joestar

(14:12:44) INFO:    [Backend]:          Starting the Download Queue
(14:12:49) ERROR:   [Gog]:              No credentials, can't get login information
(14:12:49) WARNING: [Backend]:          listUpdateableGames not implemented on Sideload Library Manager
(14:12:49) INFO:    [Gog]:              Running command: GOGDL_CONFIG_PATH=/home/incognitux/.config/heroic/gogdlConfig /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/incognitux/.config/heroic/gog_store/auth.json auth
(14:12:58) INFO:    [Gog]:              Found 1 game(s) to update
(14:12:58) INFO:    [Frontend]:         No cache found, getting data from gog...
(14:12:58) INFO:    [Gog]:              Running command: GOGDL_CONFIG_PATH=/home/incognitux/.config/heroic/gogdlConfig /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/incognitux/.config/heroic/gog_store/auth.json auth
Error occurred in handler for 'GUEST_VIEW_MANAGER_CALL': Error: ERR_FAILED (-2) loading 'https://www.epicgames.com/store/en/'
    at rejectAndCleanup (node:electron/js2c/browser_init:2:75323)
    at WebContents.stopLoadingListener (node:electron/js2c/browser_init:2:75710)
    at WebContents.emit (node:events:526:35) {
  errno: -2,
  code: 'ERR_FAILED',
  url: 'https://www.epicgames.com/store/en/'
}
```

</details>

### Add logs

```text
(14:12:24) INFO:    [Legendary]:        Legendary location: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary
(14:12:24) INFO:    [Gog]:              GOGDL location: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl
(14:12:24) INFO:    [Connection]:       Connectivity: check-online
(14:12:24) INFO:    [Connection]:       Pinging external endpoints
(14:12:26) INFO:    [Gog]:              Checking for existing gog manifests
(14:12:26) INFO:    [Backend]:          DRM module staus {
  "oimompecagnajdejgnnjijobebaeigek": {
    "status": "new",
    "title": "Widevine Content Decryption Module",
    "version": "4.10.2710.0"
  }
}
(14:12:32) WARNING: [Backend]:          Protocol already registered.
(14:12:32) WARNING: [DXVKInstaller]:    App offline, skipping possible DXVK update.
(14:12:32) INFO:    [WineDownloader]:   Updating wine versions info
(14:12:32) INFO:    [WineDownloader]:   Fetching upstream information...
(14:12:33) INFO:    [Legendary]:        Running command: LEGENDARY_CONFIG_PATH=/home/incognitux/.config/heroic/legendaryConfig/legendary /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary --version
(14:12:33) INFO:    [Gog]:              Running command: GOGDL_CONFIG_PATH=/home/incognitux/.config/heroic/gogdlConfig /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/incognitux/.config/heroic/gog_store/auth.json --version
(14:12:33) INFO:    [Nile]:             Running command: NILE_CONFIG_PATH=/home/incognitux/.config/heroic/nile_config /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/nile --version
(14:12:34) INFO:    [Connection]:       Connectivity: online
(14:12:34) INFO:    [Gog]:              Running command: GOGDL_CONFIG_PATH=/home/incognitux/.config/heroic/gogdlConfig /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/incognitux/.config/heroic/gog_store/auth.json auth
(14:12:34) INFO:    [Backend]:          User Not Found, removing it from Store
(14:12:34) INFO:    [Gog]:              Checking if login is valid
(14:12:34) INFO:    [Gog]:              Running command: GOGDL_CONFIG_PATH=/home/incognitux/.config/heroic/gogdlConfig /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/incognitux/.config/heroic/gog_store/auth.json auth
(14:12:38) INFO:    [WineDownloader]:   wine versions updated
(14:12:38) INFO:    [Frontend]:         Refreshing undefined Library
(14:12:38) WARNING: [Backend]:          refresh not implemented on Sideload Library Manager
(14:12:39) INFO:    [Legendary]:        Refreshing library...
(14:12:39) INFO:    [Gog]:              Running command: GOGDL_CONFIG_PATH=/home/incognitux/.config/heroic/gogdlConfig /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/incognitux/.config/heroic/gog_store/auth.json auth
(14:12:39) INFO:    [Backend]:          Frontend Ready
(14:12:40) INFO:    [Backend]:          AreWeAntiCheatYet data downloaded
(14:12:40) INFO:    [Backend]:          Checking for new Heroic Updates
(14:12:40) INFO:    [Backend]:          Checking for current version changelog
(14:12:44) INFO:    [Backend]:          
System Information:
CPU: 2x Intel(R) Celeron(R) CPU 3215U @ 1.70GHz
Memory: 4.01 GB (used: 2.46 GB)
GPUs:
  GPU 0:
    Name: Intel Corporation HD Graphics
    IDs: D=1606 V=8086 SD=06c1 SV=1028
    Driver: i915
OS: EndeavourOS 6.8.4-zen1-1-zen (linux)

The current system is not a Steam Deck
We are not running inside a Flatpak container

Software Versions:
  Heroic: 2.14.1 Flame Emperor Sabo
  Legendary: 0.20.34 Direct Intervention
  gogdl: 1.0.1
  Nile: 1.0.2 Jonathan Joestar

(14:12:44) INFO:    [Backend]:          Starting the Download Queue
(14:12:49) ERROR:   [Gog]:              No credentials, can't get login information
(14:12:49) WARNING: [Backend]:          listUpdateableGames not implemented on Sideload Library Manager
(14:12:49) INFO:    [Gog]:              Running command: GOGDL_CONFIG_PATH=/home/incognitux/.config/heroic/gogdlConfig /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/incognitux/.config/heroic/gog_store/auth.json auth
(14:12:58) INFO:    [Gog]:              Found 1 game(s) to update
(14:12:58) INFO:    [Frontend]:         No cache found, getting data from gog...
(14:12:58) INFO:    [Gog]:              Running command: GOGDL_CONFIG_PATH=/home/incognitux/.config/heroic/gogdlConfig /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/incognitux/.config/heroic/gog_store/auth.json auth
```

### Steps to reproduce

1. Go to Stores
2. Click on Epic Games then click on login in the "you are NOT logged in" prompt
4. See a blank page with no elements

### Expected behavior

Options to login/enter credentials appear

### Screenshots

![image](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/74859056/78eb6c0a-487f-4019-9c68-60065037b9e1)

![image](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/74859056/0008ebad-ceb6-4e57-bd76-d376dbe9ef97)

### Heroic Version

Latest Stable

### System Information

- OS [e. g. "Ubuntu"]: Endeavour OS, install `heroic-launcher-bin` using yay
- DE: KDE plasma 6 wayland session
- kernel: 6.8.4-zen1-1-zen

**Comment by AbeOwitz** [comment]
Epic login fails with both methods.  Native Arch/AUR install.  

Clicked on "Open Link" to get code, but code always fails.  Able to login to account with browser.

Terminal output:
`(13:27:15) INFO:    [Frontend]:         Called Epic Login
(13:27:15) INFO:    [Legendary]:        Logging in: LEGENDARY_CONFIG_PATH=/home/abe/.config/heroic/legendaryConfig/legendary /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary auth --code <redacted>
(13:27:22) INFO:    [Gog]:              Saved username to config file
(13:28:35) ERROR:   [Legendary]:        Failed to login with Legendary: [EPCAPI] ERROR: Login to EGS API failed with errorCode: errors.com.epicgames.account.oauth.authorization_code_not_found
[Core] ERROR: Logging in failed with InvalidCredentialsError('errors.com.epicgames.account.oauth.authorization_code_not_found'), please try again.
[cli] ERROR: Login attempt failed, please see log for details.
[cli] INFO: Testing existing login data if present...
`
Log output:
`~/.config/heroic/logs/ tail 2024-04-08T20_26_59-legendary.log
[4/8/2024, 1:27:00 PM] LEGENDARY_CONFIG_PATH=/home/abe/.config/heroic/legendaryConfig/legendary /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary --version
legendary version "0.20.34", codename "Direct Intervention"
[4/8/2024, 1:27:15 PM] LEGENDARY_CONFIG_PATH=/home/abe/.config/heroic/legendaryConfig/legendary /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary auth --code <redacted>
[cli] INFO: Testing existing login data if present...
[EPCAPI] ERROR: Login to EGS API failed with errorCode: errors.com.epicgames.account.oauth.authorization_code_not_found
[Core] ERROR: Logging in failed with InvalidCredentialsError('errors.com.epicgames.account.oauth.authorization_code_not_found'), please try again.
[cli] ERROR: Login attempt failed, please see log for details.`

**Comment by Habarug** [comment]
Same here, first time installing Heroic. Also tried uninstalling and installing 2.14.0 instead, but it acts the same way.

Fedora 39, Gnome 35, Heroic installed from COPR repo.

**Comment by kenderipa** [comment]
Same also on my side.

**Comment by CommandMC** [comment]
The linked PR should resolve the issue. You can either use Heroic built from there, or work around the issue by navigating to "Manage Accounts" in the sidebar directly.

@AbeOwitz Your issue sounds different than everything else here. The log would suggest that you didn't paste in your authorization code into the input box. Assuming you're still not able to login after doing this, please open up your own new issue.

**Comment by h0m3** [comment]
Can confirm, the "Stores" page just show a Blank login page after the "You're not logged in" warning. If you go to Manage Accounts or the Login button you can login.

**Comment by arielj** [comment]
> same here for me

use the Manage Accounts element in the sidebar instead of the link from the store

**Comment by jaspov** [comment]
Just adding my experience on steamdeck. Could not install a game in heroic (launched from game mode). Version 2.14.1. Error quite cryptic. 

Launched heroic and while browsing through settings got the “not logged on message”. Clicked and got blank screen as described. 

As suggested above went to manage accounts to log on. However epic captcha failed to work. My mouse could only be used on part of the heroic screen. Managed to select pictures and using tab get to next button but it failed validation.  

Proceeded to desktop mode and ther could log on again, all resolved. 

Just wanted to add that this is all quite confusing, the initial error not pointing to “not being logged on”. Then log on failing with black screen. Alternative not working in desktop mode. Hope it gets fixed or error messages improved to point to solution. Thanks for all the great work!

**Comment by CommandMC** [comment]
> Launched heroic and while browsing through settings got the “not logged on message”. Clicked and got blank screen as described.

This will be resolved in the next release (whenever that'll be).

> However epic captcha failed to work. My mouse could only be used on part of the heroic screen. Managed to select pictures and using tab get to next button but it failed validation.

> Proceeded to desktop mode and ther could log on again, all resolved.

This is up to Epic, nothing we can do about their captcha verification and/or accessibility. We just open the log in page in Heroic and automatically parse the response once the login flow is complete.
